src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs

```diff
@@ -89,27 +90,31 @@ public static string DiffFiles(string file1Path, string file2Path, ITestOutputHe
 
         public static string GetBaselineFilePath(string baselineFileName) => Path.Combine(GetAssetsDirectory(), "baselines", baselineFileName);
 
-        public static string RemoveNetTfmPaths(string source)
-        {
-            string pathSeparator = Regex.Escape(Path.DirectorySeparatorChar.ToString());
-            Regex netTfmRegex = new($"{pathSeparator}net[1-9]+\\.[0-9]+{pathSeparator}");
-            return netTfmRegex.Replace(source, $"{Path.DirectorySeparatorChar}{NetTfmPlaceholder}{Path.DirectorySeparatorChar}");
-        }
-
         public static string RemoveRids(string diff, bool isPortable = false) =>
             isPortable ? diff.Replace(Config.PortableRid, "portable-rid") : diff.Replace(Config.TargetRid, "banana-rid");
 
         public static string RemoveVersions(string source)
         {
+            // Remove version numbers for examples like "roslyn4.1", "testhost-1.0", "net8.0", and "netstandard2.1".
+            string result = Regex.Replace(source, $@"(roslyn|testhost-|netstandard|net)\d+\.\d+", match =>
+            {
+                string wordPart = match.Groups[1].Value;
+                return $"{wordPart}x.y";
+            });
+
             // Remove semantic versions
             // Regex source: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+            // The regex from https://semver.org has been modified to account for the following:
+                // - The version should be preceded by a path separator, '.', '-', or '/'
+                // - The version should match a release identifier that begins with '.' or '-'
+                // - The version may have one or more release identifiers that begin with '.' or '-'
+                // - The version should end before a path separator, '.', '-', or '/'
             Regex semanticVersionRegex = new(
-                $"(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)"
-                + $"(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))"
-                + $"?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?");
-            string result = semanticVersionRegex.Replace(source, VersionPlaceholder);
-
-            return RemoveNetTfmPaths(result);
+                @"(?<=[./-])(0|[1-9]\d*)\.(0|[1-9]\d*)(\.(0|[1-9]\d*))+" +
```

Comment:
It looks like RemoveNetTmfPaths was only used in RemoveVersions, and RemoveVersions is only used in two other test cases, besides ArtifactsSizeTest, throughout the codebase.

The first test is PoisonTests, for which the output is PoisonUsage.txt. PoisonUsage.txt now contains no text that would be affected by this change since https://github.com/dotnet/installer/pull/17159.

The second test is SdkContentTests. From what I can tell, this test is currently failing in the pipeline for issues unrelated to my changes (i.e. these changes don't affect the baseline). That said, I'd like to wait for a green build to verify this before pushing more changes.

Reply:
SDK diff tests are green now so you should be good to go.

Reply:
@mthalman I pulled and updated my baselines from the last green run of the SDK diff tests, but I'm continuing to experience issues with failing tests.

I've confirmed that MsftToSbSdkFiles.diff needs to be updated because of the Regex logic for changing "netstandard2.1" to "netstandardx.y", but I am seeing[ lines in the output of the failing test that are unrelated to this logic](https://dev.azure.com/dnceng/internal/_build/results?buildId=2258210&view=logs&j=accab69a-2e25-5427-de75-1a40331eaf8f&t=7e2438f2-f0a8-51dd-3612-7ec3fb2768b2&l=109). As such, I'm unsure of why the baselines have these other differences and if this is an issue on my end.

Reply:
Ah, you're targeting the 8.0 branch for the test. That requires this update to the baseline: https://github.com/dotnet/installer/pull/17282. That's the only other change that's needed, AFAIK.

Reply:
Fixed the discrepancies in [this commit](https://github.com/dotnet/installer/pull/17233/commits/e7de907edbb6ce9fb5555254e0c2bf0ea32ac06c). The changes involved:
- Removing "testhost-..." from the regex pattern
- Adding the pathSeparator back into the regex for non-semantic versions
- Updating the NetTfmPlaceholder and NetTfmPlaceholderMatchingPattern to be generalized for non-semantic versions
- Updating MsftToSbSdkFiles.diff to account for including netstandard in the version replacement